File: _posts/arrays/single_number/ts/solution_map.ts
export const singleNumberMap = (nums: number[]): number => {
    const countMap: Record<number, number> = {}

    nums.forEach((num) => {
        const curr = countMap[num] ?? 0
        countMap[num] = curr + 1
    })

    const entries = Object.entries(countMap)

    for (let i = 0; i < entries.length; i++) {
        const [num, count] = entries[i]
        if (count === 1) {
            return Number(num)
        }
    }

    return 0
}

File: content/blog/single_number/go/solution_map.go
package single_number

func singleNumberMap(nums []int) int {
	countMap := make(map[int]int)

	for _, num := range nums {
		countMap[num] += 1
	}

	for num, count := range countMap {
		if count == 1 {
			return num
		}
	}

	return 0
}
